apps/teamsjs-diagnostic-app/src/components/sample/Scenario1.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useState, useEffect } from 'react';
+import './Scenario1.css';
+import { captureConsoleLogs } from './LoggerUtility';
+import { app } from '@microsoft/teams-js';
+import { registerOnResume } from '../../apis/AppApi';
+import { authenticateUser } from '../../apis/AuthenticationStart';
+import apiComponents from './ApiComponents';
+
+type Log = string;
+
+interface Scenario1Props {
+  showFunction?: boolean;
+}
+
+export function Scenario1({ showFunction }: Scenario1Props) {
+  const [logStatements, setLogStatements] = useState<Log[]>([]);
+  const [searchQuery, setSearchQuery] = useState('');
+
+  useEffect(() => {
+    const filteredLogs: Log[] = [];
+    captureConsoleLogs((log: string) => {
+      if (!log.includes('Get basic user info from SSO token')) {
+        filteredLogs.push(log);
+        setLogStatements([...filteredLogs]);
+        localStorage.setItem('logStatements', JSON.stringify(filteredLogs));
+      }
+    });
+
+    app.initialize();
+  }, []);
+
+  const runAppInitializationScenario = async () => {
+    try {
+      console.log('Running App Initialization Scenario...');
+      console.log('Attempting to register on resume handler...');
+      await registerOnResume();
+      console.log('Attempting to authenticate user...');
+      const authSuccess = await authenticateUser();
+      if (authSuccess) {
+        console.log('App Initialization Scenario successfully completed');
+      } else {
+        console.log('User not authenticated');
+        showSignInPopup();
+      }
+    } catch (error: any) {
+      console.log(`App initialization scenario failed. ${error.message}`);
+    }
+  };
+
+  const showSignInPopup = () => {
+    console.log('Showing sign-in popup...');
+  };
+
+  useEffect(() => {
+    const storedLogs = localStorage.getItem('logStatements');
```

Comment:
I don't think you need to use localStorage here. LocalStorage is always a little bit of a security/privacy concern because technically whatever you write into local storage could be there for a very long time, and if the user is on a shared computer of some sort another user could open local storage and see what's in there.

It might be better to store the collected log statements in memory inside of your app instead of local storage. That will make sure that if a user refreshes the page or navigates away their logs won't be stored somewhere that they don't know about.

Reply:
I tried doing that but the issue I am facing is that I have certain logs on my console, and then I would click my "Run Scenario" button that deals with authentication, the page reloads and deletes all the logs, but i want them to still be there so it can show a continuous log. The only way I was able to figure out how to fix this was with localStorage, but I will continue looking into other ways.